pkg/controller/cluster/controller.go

```diff
@@ -59,33 +59,35 @@ func (c *ClusterControllerImpl) Init(arguments sharedinformers.ControllerInitArg
 
 // Reconcile handles enqueued messages. The delete will be handled by finalizer.
 func (c *ClusterControllerImpl) Reconcile(cluster *clusterv1.Cluster) error {
-	name := cluster.Name
+	// Deep-copy otherwise we are mutating our cache.
+	clusterCopy := cluster.DeepCopy()
+	name := clusterCopy.Name
 	glog.Infof("Running reconcile Cluster for %s\n", name)
 
 	if !cluster.ObjectMeta.DeletionTimestamp.IsZero() {
```

Comment:
Looking at the machine controller, it seems like this line and line 81 should also use clusterCopy instead of cluster. Is there a reason to use the original object in these cases?

Reply:
There was not, fixed in latest. Thanks!